spec/models/orchestration_template_cfn_spec.rb

```diff
@@ -141,4 +147,35 @@ def assert_hidden_length_pattern(parameter)
       expect(template.validate_format).not_to be_nil
     end
   end
+
+  describe '#deployment_options' do
+    it 'generates deployment options for AWS' do
+      options = subject.deployment_options('ManageIQ::Providers::Amazon::CloudManager')
+      assert_deployment_option(options[0], "tenant_name", :OrchestrationParameterAllowedDynamic, true)
+      assert_deployment_option(options[1], "stack_name", :OrchestrationParameterPattern, true)
+      assert_deployment_option(options[2], "stack_onfailure", :OrchestrationParameterAllowed, false)
+      assert_deployment_option(options[3], "stack_timeout", 'integer', nil, false)
+      assert_deployment_option(options[4], "stack_notifications", 'text', nil, false)
+      assert_deployment_option(options[5], "stack_capabilities", :OrchestrationParameterAllowed, false)
+      assert_deployment_option(options[6], "stack_resource_types", 'text', nil, false)
+      assert_deployment_option(options[7], "stack_role", 'string', nil, false)
+      assert_deployment_option(options[8], "stack_tags", 'text', nil, false)
+      assert_deployment_option(options[9], "stack_policy", 'text', nil, false)
+    end
+
+    it 'generates deployment options for OpenStack' do
+      options = subject.deployment_options('ManageIQ::Providers::Openstack::CloudManager')
+      assert_deployment_option(options[0], "tenant_name", :OrchestrationParameterAllowedDynamic, true)
+      assert_deployment_option(options[1], "stack_name", :OrchestrationParameterPattern, true)
+      assert_deployment_option(options[2], "stack_onfailure", :OrchestrationParameterAllowed, false)
+      assert_deployment_option(options[3], "stack_timeout", 'integer', nil, false)
+    end
+  end
+
+  def assert_deployment_option(option, name, data_type = 'string', constraint_type, required)
```

Comment:
@bzwei While I realize that having the middle parameter defaulted works I feel like it makes it confusing for people reading the code.  Since this is all new can we just push the defaulted parameter to the end of the list.